### ISSUE TITLE
Doorbell press event support

### ISSUE THREAD

**Issue by ruben-janssens** [QuantumEntangledAndy/neolink#339]
**Is your feature request related to a problem? Please describe.**
I am using Neolink to connect to the new WiFi battery doorbell.  Everything works like expected (after some fiddling of course) but now I need to know when someone presses the bell.

I would implement it myself but then I would need to learn Rust from scratch :)

**Comment by QuantumEntangledAndy** [comment]
It's a nice enhancement and I wouldn't mind tackling it. But I'd need access to the wireshark packets to reverse engineer the protocol

**Comment by ruben-janssens** [comment]
If you can provide some steps on how I should setup an environment to capture those packets I can provide them to you.

**Comment by QuantumEntangledAndy** [comment]
Oh we have CMD 33 already done. It will pop up as a motion alert but we filter it out.

**Comment by QuantumEntangledAndy** [comment]
How would you expect it to be integrated? Are you using the mqtt to read events?

**Comment by ruben-janssens** [comment]
Yes I use MQTT to receive everything on HomeAssistant so that would be ideal :)

**Comment by QuantumEntangledAndy** [comment]
I'm working on a build for this to test. Are you using docker or bare metal neolink?

**Comment by ruben-janssens** [comment]
I use the neolink package on a debian LXC in proxmox. Thanks for the quick implementation!

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by QuantumEntangledAndy** [comment]
Ok I have the build happening [here](https://github.com/QuantumEntangledAndy/neolink/actions/runs/12947382929) hopefully it will work and then the binaries will be available

**Comment by ruben-janssens** [comment]
I'm currently not home so I can test this in 3 ish hours.

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by QuantumEntangledAndy** [comment]
Build will take an hour anyways so no rush

**Comment by QuantumEntangledAndy** [comment]
Ok build of the binaries are here https://github.com/QuantumEntangledAndy/neolink/actions/runs/12947507860

Docker is here quantumentangledandy/neolink:feat-visitor

**Comment by ruben-janssens** [comment]
Talk about perfectly timed,  will test them now

**Comment by ruben-janssens** [comment]
Do I have to modify any config to add the visitor functionality ?
Like  for example in mqqt features ?

**Comment by ruben-janssens** [comment]
Alright got the event! A small problem dho

```
Message 138438 received on neolink/Doorbell/status/visitor at 15:13:

visitor
```

Neolink has send the event almost 140000 times

**Comment by QuantumEntangledAndy** [comment]
https://github.com/QuantumEntangledAndy/neolink/actions/runs/12961197957 next build

Should fix the repeating notifications, couldnt replicate the motion stopping on mine but I don't have ai or visitor

**Comment by ruben-janssens** [comment]
I will be able to test the next build in a week


________________________________

**Comment by ruben-janssens** [comment]
I see that the issue is now fixed. Thanks! But for some reason motion events are still not pushed through.
I have enabled debug mode but that also gives nothing.

**Comment by QuantumEntangledAndy** [comment]
Is the motion on your camera PIR?

**Comment by QuantumEntangledAndy** [comment]
Reolink has some different ways to detect motion. 
- One of the ways is PiR into AI
- another is PIR only (can't detect that any more since we lost push notifications)
- another is detecting changed pixels between frames

The motion detection in Neolink is for the third kind, changed frames

Did it work before? I'd like to get a wireshark dump so I can examine the packets and see what is coming from the cameras can you provide one?

**Comment by ruben-janssens** [comment]
Ah then I think it is not working because of the lost push notifications. I have it setup to disconnect when no motion is detected after a certain amount of time because it is a battery powered doorbell.

**Comment by QuantumEntangledAndy** [comment]
Since there's no push notifications it won't reconnect by itself anymore.

P.s. do you have an NVR? I think I can reverse engineer their protocol to hook into its notification system. But I don't have one myself to find out

**Comment by ruben-janssens** [comment]
Sadly I don't have the NVR or the home hub. I have thought about buying one of them if neolink didn't work out.
Maybe for the greater good I should buy one ...

If we can solve it in 30 days I could theoretically send it back even.

**Comment by QuantumEntangledAndy** [comment]
I'm not sure about the less than 30 days thing. (Work is always busy and would require a lot of effort to reverse engineer)

I'm hoping to find something to help or maybe try other experiments.

**Comment by ruben-janssens** [comment]
Hi @QuantumEntangledAndy, just wanted to let you know that I will be buying the Home Hub. Just waiting for a sale price.
Once I have it I will get back in touch

**Comment by ruben-janssens** [comment]
@QuantumEntangledAndy I'm in luck and got one at a discount :) Whenever you have some free time could you provide me with a setup to gather the packets between the home hub and the battery doorbell

**Comment by koproductions-code** [comment]
Hey there. Have you made any progress on this? Also using a battery doorbell and would love to integrate it into homeassistant to play chimes on my smart speakers on button press.

**Comment by ruben-janssens** [comment]
Hi @koproductions-code @QuantumEntangledAndy I don't know if it's interesting but I'm trying to reverse engineer the connection in https://github.com/starkillerOG/reolink_aio/issues/125 because my python understanding is better than Rust

**Comment by sh00t2kill** [comment]
I know this is a little stale; but any updates ? Anything you need in order to get this working ? Happy to help as have just got this all up and running.

**Comment by ruben-janssens** [comment]
Hi @sh00t2kill ,  as mentioned in https://github.com/starkillerOG/reolink_aio/issues/125 I haven't had much time lately.
I have now spend another few hours trying to decode the communication between the home hub and reolink battery device but to no avail. There is some other form of encryption being used for the events send from the reolink device to the home hub. I have found a password, accessKey, key and device ids. Also some form of structure in the packets but thats it.
If you would like to help maybe you can take a look at it  yourself. Unfortunately I can't share any of the keys because the home hub uses static values.
